compass/mapper/plugins/basic/basic.go

```diff
@@ -74,38 +80,67 @@ func (m *Mapper) Map(evidence api.RawEvidence, scope mapper.Scope) (api.Complian
 		controlData := m.buildControlDataMap(catalog)
 
 		// Look up policy in procedures
-		if procedureInfo, ok := proceduresById[evidence.PolicyId]; ok {
+		if procedureInfo, ok := proceduresById[evidence.PolicyRuleId]; ok {
 
 			// Look up control data
 			if ctrlData, ok := controlData[procedureInfo.ControlID]; ok {
 				compliance := api.Compliance{
-					Catalog:      catalogId,
-					Control:      procedureInfo.RequirementID,
-					Requirements: m.extractRequirements(ctrlData.Mappings),
-					Standards:    m.extractStandards(ctrlData.Mappings),
-					Category:     ctrlData.Category,
-					Remediation:  &procedureInfo.Documentation,
+					Control: api.ComplianceControl{
+						Id:                     procedureInfo.RequirementID,
+						Category:               ctrlData.Category,
+						RemediationDescription: &procedureInfo.Documentation,
+						CatalogId:              catalogId,
+					},
+					Frameworks: api.ComplianceFrameworks{
+						Requirements: m.extractRequirements(ctrlData.Mappings),
+						Frameworks:   m.extractStandards(ctrlData.Mappings),
+					},
+					Status:           status,
+					EnrichmentStatus: api.ComplianceEnrichmentStatusSuccess,
 				}
 
-				return compliance, api.Status{Title: status, Id: &statusId}
+				return compliance
+			} else {
+				log.Printf("WARNING: Control data not found for control ID %s in catalog %s for policy %s", procedureInfo.ControlID, catalogId, evidence.PolicyRuleId)
+				failureReasons = append(failureReasons, "control data not found")
 			}
+		} else {
+			log.Printf("WARNING: Policy rule %s not found in procedures for catalog %s", evidence.PolicyRuleId, catalogId)
+			failureReasons = append(failureReasons, "policy rule not found")
 		}
 	}
 
-	return api.Compliance{}, api.Status{Title: status, Id: &statusId}
+	// Log final failure if no mapping was found
+	if len(failureReasons) > 0 {
+		log.Printf("WARNING: Failed to map policy %s from engine %s. Reasons: %v", evidence.PolicyRuleId, evidence.PolicyEngineName, failureReasons)
+	}
+
+	return api.Compliance{
+		Status: api.UNKNOWN,
+		Control: api.ComplianceControl{
+			Id:        "UNMAPPED",
+			CatalogId: "UNMAPPED",
+			Category:  "UNCATEGORIZED",
+		},
+		EnrichmentStatus: enrichmentStatus,
```

Comment:
It is a nit, but this variable is defined at the begging and only used here. Maybe we could use `api.ComplianceEnrichmentStatusUnmapped` directly since the next definition is already consuming directly from `api`.